### PR TITLE
ddl: set correct startHandle when add indices meets some errors (#6897)

### PR DESCRIPTION
**Pull request by winkyao**
This is a cherry-pick from #6897

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

**Comment by tiancaiamao** [comment]
LGTM

**Comment by tiancaiamao** [comment]
/run-all-tests tidb-test=release-2.0 tikv=release-2.0 pd=release-2.0

File: ddl/index.go [hunk 1]
```diff
@@ -637,8 +639,15 @@ func (w *addIndexWorker) run() {
 		if !more {
 			break
 		}
-
 		log.Debug("[ddl-reorg] got backfill index task:#v", task)
+
+		// gofail: var mockAddIndexErr bool
+		//if mockAddIndexErr && !gofailMockAddindexErrOnceGuard && w.id == 0 {
```
Comment (coocood) [inline ddl/index.go]: We should check the `w.id` first.

**Comment by coocood** [comment]
LGTM

**Comment by coocood** [comment]
/run-unit-test tidb-test=release-2.0 tikv=release-2.0 pd=release-2.0

**Pull request merged**